packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasSummary.ts

```diff
@@ -0,0 +1,252 @@
+  const [gasSummaryStatus, setGasSummaryStatus] =
+    useState<GasEstimationStatus>('loading')
+  const [gasSummary, setGasSummary] = useState<UseGasSummaryResult>(
+    INITIAL_GAS_SUMMARY_RESULT
+  )
+
+  const amountBigNumber = useMemo(() => {
+    try {
+      const amountSafe = amount || '0'
+
+      if (token) {
+        return utils.parseUnits(amountSafe, token.decimals)
+      }
+
+      return utils.parseUnits(amountSafe, nativeCurrency.decimals)
+    } catch (error) {
+      return constants.Zero
+    }
+  }, [amount, token, nativeCurrency])
+
+  const parentChainGasPrice = useGasPrice({ provider: parentChainProvider })
+  const childChainGasPrice = useGasPrice({ provider: childChainProvider })
+
+  // Debounce the amount, so we run gas estimation only after the user has stopped typing for a bit
+  const amountDebounced = useDebouncedValue(amountBigNumber, 1_500)
+
+  const [result, setResult] = useState<GasEstimationResult>(
+    INITIAL_GAS_ESTIMATION_RESULT
+  )
+
+  const estimateGas = useCallback(async () => {
+    if (!walletAddress) {
+      return
+    }
+
+    const estimateGasFunctionParams = {
+      amount: amountDebounced,
+      address: walletAddress,
+      l2Provider: childChainProvider
+    }
+
+    let estimateGasResult: GasEstimationResult = INITIAL_GAS_ESTIMATION_RESULT
+
+    try {
+      setGasSummaryStatus('loading')
+
+      if (isDepositMode) {
+        estimateGasResult = token
+          ? await depositTokenEstimateGas({
+              ...estimateGasFunctionParams,
+              l1Provider: parentChainProvider,
+              erc20L1Address: token.address
+            })
+          : await depositEthEstimateGas({
+              ...estimateGasFunctionParams,
+              l1Provider: parentChainProvider
+            })
+      } else {
+        const partialEstimateGasResult = await withdrawInitTxEstimateGas({
+          ...estimateGasFunctionParams,
+          erc20L1Address: token ? token.address : undefined
+        })
+
+        estimateGasResult = {
+          ...partialEstimateGasResult,
+          estimatedL2SubmissionCost: constants.Zero
+        }
+      }
+
+      setResult(estimateGasResult)
+      setGasSummaryStatus('success')
+    } catch (error) {
+      console.error(error)
+      setGasSummaryStatus('error')
+    }
+  }, [
+    // Re-run gas estimation when:
+    isDepositMode, // when user switches deposit/withdraw mode
+    amountDebounced,
+    token, // when the token changes
+    walletAddress, // when user switches account or if user is not connected
+    setGasSummaryStatus,
+    parentChainProvider,
+    childChainProvider
+  ])
+
+  // Estimated L1 gas fees, denominated in Ether, represented as a floating point number
+  const estimatedL1GasFees = useMemo(() => {
+    const gasPrice = isDepositMode ? parentChainGasPrice : childChainGasPrice
+    return parseFloat(utils.formatEther(result.estimatedL1Gas.mul(gasPrice)))
```

Comment:
this was introduced due to merge conflicts as this branch is very old and the refactoring was done later

addressed here https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1490

since it's 0 atm it doesn't matter